index.js

```diff
@@ -221,78 +228,112 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
-    }
-    return typeof x === 'function' ? x : null;
+  //  prefix :: String
+  var prefix = 'fantasy-land/';
+
+  //  funcProp :: (String, a) -> Nullable Function
+  function funcProp(prop, x) {
+    return x != null && typeof x[prop] === 'function' ? x[prop] : null;
   }
 
-  //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
+  //  functionImpl :: (String, String) -> Nullable Function
+  function functionImpl(type, name) {
+    return implementations[type] && implementations[type][name] || null;
   }
 
-  //  implPath :: Array String -> Nullable Function
-  function implPath(path) {
-    return _funcPath(false, path, implementations);
+  //  methodImpl :: (String, String) -> Nullable Function
+  function methodImpl(type, name) {
+    return implementations[type] &&
+           implementations[type].prototype[name] ||
+           null;
   }
 
   //  functionName :: Function -> String
-  var functionName = has('name', function f() {}) ?
+  var functionName = has('name', Function) ?
     function functionName(f) { return f.name; } :
     /* istanbul ignore next */
     function functionName(f) {
       var match = /function (\w*)/.exec(f);
       return match == null ? '' : match[1];
     };
 
+  //  getFunction :: (String, TypeRep a) -> Nullable Function
+  function getFunction(_name, typeRep) {
+    var name = prefix + _name;
+    return (funcProp(name, typeRep) ||
+             typeof typeRep === 'function' &&
+             functionImpl(functionName(typeRep), name)) ||
+           null;
+  }
+
+  function isPrototype(o) {
+    return o != null && o.constructor != null && o.constructor.prototype === o;
+  }
+
+  //  hasMethod :: (String, Any) -> Boolean
+  function hasMethod(_name, value) {
+    var name = prefix + _name;
+    return methodImpl(type(value), name) != null ||
+           isPrototype(value) === false &&
+           funcProp(name, value) != null;
+  }
+
+  //  makeFunction :: String -> TypeRep a -> Function
+  function makeFunction(_name) {
+    return function(typeRep) {
+      return getFunction(_name, typeRep);
+    };
+  }
```

Comment:
`makeFunction` is just `getFunction` curried. Does having access to the uncurried function make a significant difference to performance? If so, could we name this function `getFunctionCurried` or similar?